winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp

```diff
@@ -1061,10 +1067,12 @@ TEST_F(ScenarioCppWinrtGpuTest, DISABLED_Scenario22_ImageBindingAsGPUTensor)
     );
 
     // Create the GPU upload buffer.
+    CD3DX12_HEAP_PROPERTIES props(D3D12_HEAP_TYPE_UPLOAD);
+    auto buffer = CD3DX12_RESOURCE_DESC::Buffer(bufferbytesize);
```

Comment:
this was a change for warnings?   or an unrelated change?

Reply:
warning C4238: nonstandard extension used:
 class rvalue used as lvalue
[https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4238?view=vs-2019](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4238?view=vs-2019)
For compatibility with previous versions of Visual C++, Microsoft extensions (/Ze) allow you to use a class type as an rvalue in a context that implicitly or explicitly takes its address. In some cases, such as the example below, this can be dangerous.

---
In reply to: [353328719](https://github.com/microsoft/onnxruntime/pull/2526#discussion_r353328719) [](ancestors = 353328719)